From f7b738ff2d793691589b46d5de4edd119fa15e9f Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 28 May 2013 10:03:07 +0300
Subject: [PORT FROM MAIN] Telephony: Fix for SIM pin dialog shown on sim not ready

BZ: 107119

When the airplane mode is activated, sim pin dialog
is shown even when the app state is transitioned from
RIL_APPSTATE_PIN to RIL_APPSTATE_UNKNOWN. When the sim
is not accessible in radio off, sim state changed intent
can be received with app state set to unknown. Upon,
sim app state received with RIL_APPSTATE_UNKNOWN, framework
notifies the sim state as UNKNOWN. Policy framework ignores
the UNKNOWN state thus resulting in SIM pin dialog shown
even when sim is not in ready state.

Update the sim state as NOT_READY when the received
app_state is APPSTATE_UNKNOWN/APPSTATE_DETECTED.

Change-Id: I83b9daf19b2d3e059ddfe0d27667b9ca43c39134
Orig-Change-Id: Idc02281b5c437d50a3fbf8a69164ee4d3af8394a
Category: device enablement
Domain: Telephony-Modem-Intel
Origin: internal
Upstream-Candidate: no,proprietary
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 .../internal/telephony/uicc/IccCardProxy.java      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/IccCardProxy.java b/src/java/com/android/internal/telephony/uicc/IccCardProxy.java
index c512ecf..0f28583 100644
--- a/src/java/com/android/internal/telephony/uicc/IccCardProxy.java
+++ b/src/java/com/android/internal/telephony/uicc/IccCardProxy.java
@@ -290,7 +290,7 @@ public class IccCardProxy extends Handler implements IccCard {
         switch (mUiccApplication.getState()) {
             case APPSTATE_UNKNOWN:
             case APPSTATE_DETECTED:
-                setExternalState(State.UNKNOWN);
+                setExternalState(State.NOT_READY);
                 break;
             case APPSTATE_PIN:
                 setExternalState(State.PIN_REQUIRED);
-- 
1.7.4.1

